yarn-project/types/src/contract_data.ts

```diff
@@ -3,6 +3,25 @@ import { BufferReader, serializeToBuffer } from '@aztec/circuits.js/utils';
 
 export { BufferReader } from '@aztec/circuits.js/utils';
 
+/**
+ * Used for retrieval of contract data (A3 address, portal contract address, bytecode)
+ */
+export interface ContractDataSource {
+  /**
+   * Lookup the L2 contract data for this contract.
+   * Contains information such as the ethereum portal address and bytecode.
+   * @param contractAddress - The contract data address.
+   * @returns The portal address (if found).
+   */
+  getL2ContractData(contractAddress: AztecAddress): Promise<ContractData | undefined>;
```

Comment:
I understand `ContractData` as is today does not return the bytecode, right? What I'm looking for is an interface that, given contract address and function selector, returns the bytecode for that function.

Reply:
thx for clarifying that as the task says "retrieval of contract data". I'm currently implementing the bytecode part now, I construed the requirement as returning the bytecode of the entire contract

Reply:
Lets add an additional method on the interface to provide a specific function